packages/autocomplete-core/src/onInput.ts

```diff
@@ -117,7 +102,7 @@ export function onInput<TItem>({
           setSuggestions(suggestions as any);
           setIsOpen(
             nextState.isOpen ??
-              (query.length >= props.minLength &&
+              (props.openOnFocus ||
```

Comment:
`openOnFocus` should always be paired with a query length check: `(query.length  === 0 && props.openOnFocus) || /* ... */`.